src/framework/rendering/Model3D.cpp

```diff
@@ -0,0 +1,268 @@
+Model3D::Model3D()
+{
+   m_vaoId = 0;
+   for(int i = 0; i < vertexAttributeLast; i++)
+      m_BufferIds[i] = 0;
+   
+   m_vertexCount = 0;
+}
+
+//---------------------------------------------------------------------------------------------------
+Model3D::~Model3D()
+{
+   UnBindVAO();
+   glDeleteVertexArrays(1, &m_vaoId);
+
+   glBindBuffer(GL_ARRAY_BUFFER, 0);
+   glDeleteBuffers(vertexAttributeLast, m_BufferIds);
+}
+
+//---------------------------------------------------------------------------------------------------
+void Model3D::BindVAO()
+{
+   glBindVertexArray(m_vaoId);
+
+   if(m_BufferIds[vertexAttributePosition] != 0)
+      glEnableVertexAttribArray(vertexAttributePosition);
+   
+   if(m_BufferIds[vertexAttributeColor] != 0)
+      glEnableVertexAttribArray(vertexAttributeColor);
+
+   if(m_BufferIds[vertexAttributeNormal] != 0)
+      glEnableVertexAttribArray(vertexAttributeNormal);
+
+   if(m_BufferIds[vertexAttributeTexture] != 0)
+      glEnableVertexAttribArray(vertexAttributeTexture);
+
+   if(m_BufferIds[vertexAttributeTangent] != 0)
+      glEnableVertexAttribArray(vertexAttributeTangent);
+}
+
+//---------------------------------------------------------------------------------------------------
+void Model3D::UnBindVAO()
+{
+   if(m_BufferIds[vertexAttributePosition] != 0)
+      glDisableVertexAttribArray(vertexAttributePosition);
+
+   if(m_BufferIds[vertexAttributeColor] != 0)
+      glDisableVertexAttribArray(vertexAttributeColor);
+
+   if(m_BufferIds[vertexAttributeNormal] != 0)
+      glDisableVertexAttribArray(vertexAttributeNormal);
+
+   if(m_BufferIds[vertexAttributeTexture] != 0)
+      glDisableVertexAttribArray(vertexAttributeTexture);
+
+   if(m_BufferIds[vertexAttributeTangent] != 0)
+      glDisableVertexAttribArray(vertexAttributeTangent);
+
+   glBindVertexArray(0);
+   glBindBuffer(GL_ARRAY_BUFFER, 0);
+}
+
+
+//---------------------------------------------------------------------------------------------------
+void Model3D::Translate(vec3& translations)
+{
+   modelMatrix *= glm::translate(translations);
+}
+
+//---------------------------------------------------------------------------------------------------
+void Model3D::Rotate(vec3& rotations)
+{
+   modelMatrix *= glm::rotate(rotations.x, vec3(1.0f, 0.0f, 0.0f));
+   modelMatrix *= glm::rotate(rotations.y, vec3(0.0f, 1.0f, 0.0f));
+   modelMatrix *= glm::rotate(rotations.z, vec3(0.0f, 0.0f, 1.0f));
+}
+
+//---------------------------------------------------------------------------------------------------
+void Model3D::ResetPosition()
+{
+   modelMatrix = mat4();
+}
+
+//---------------------------------------------------------------------------------------------------
+mat4 Model3D::GetModelMatrix()
+{
+   return modelMatrix;
+}
+
+//---------------------------------------------------------------------------------------------------
+bool Model3D::CreateVertexBuffer(vector<vec3>& data)

```

Comment:
Looks like CreateXXXBuffer methods have unique types. You might want to think of combining these to a single method. Maybe by passing the enumeration describing type of buffer to be created.

However I'm good with the current approach too.